source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveColumnSetRenderer.cs

```diff
@@ -129,29 +134,48 @@ public static FrameworkElement Render(AdaptiveColumnSet columnSet, AdaptiveRende
                     width = column.Size?.ToLower();
 #pragma warning restore CS0618 // Type or member is obsolete
                 if (width == null || width == AdaptiveColumnWidth.Stretch.ToLower())
-                    uiColumnSet.ColumnDefinitions.Add(new ColumnDefinition() { Width = new GridLength(1, GridUnitType.Star) });
+                {
+                    ColumnDefinition columnDefinition = new ColumnDefinition() { Width = new GridLength(1, GridUnitType.Star) };
+                    tag.NotAutoWidthColumnDefinition = columnDefinition;
+                    tag.ViewIndex = uiColumnSet.ColumnDefinitions.Count;
+                    uiColumnSet.ColumnDefinitions.Add(columnDefinition);
```

Comment:
can't this more cleanly be written by generating the `ColumnDefinition` for each situation, and then adding it in one spot?